aziotctl/src/internal/check/checks/host_connect_iothub.rs

```diff
@@ -68,17 +68,35 @@ impl HostConnectIotHub {
     ) -> Result<CheckResult> {
         use aziot_identityd_config::ProvisioningType;
 
-        let iothub_hostname = match &shared.cfg.iothub_hostname {
-            Some(s) => s,
+        let identityd_config = unwrap_or_skip!(&cache.cfg.identityd);
+
+        let (iothub_hostname, nested) = match &shared.cfg.iothub_hostname {
+            Some(s) => (s, false),
             None => {
-                let iothub_hostname = match &unwrap_or_skip!(&cache.cfg.identityd)
-                    .provisioning
-                    .provisioning
-                {
+                let (iothub_hostname, nested) = match &identityd_config.provisioning.provisioning {
                     ProvisioningType::Manual {
                         iothub_hostname, ..
-                    } => iothub_hostname,
+                    } => {
+                        // use `local_gateway_hostname` if available
+                        (
+                            identityd_config
+                                .provisioning
+                                .local_gateway_hostname
+                                .as_ref()
+                                .unwrap_or(iothub_hostname),
+                            true,
```

Comment:
shouldn't this be `identityd_config.provisioning.local_gateway_hostname.is_some()`?

Reply:
Good catch it was a bug